lib/discourse_chat/provider/slack/slack_provider.rb

```diff
@@ -94,6 +112,11 @@ def self.send_via_api(post, channel, message)
       channel: message[:channel].gsub('#', ''),
       attachments: message[:attachments].to_json
     }
+    if message.key?(:thread_ts)
+      data[:thread_ts] = message[:thread_ts]
+    elsif match = slack_thread_regex.match(post.cooked)
+      post.topic.set_slack_thread_id(match.captures[1])
```

Comment:
I think you will need to call `topic.save!` or `topic.save_custom_fields` here, so that the custom field change is persisted to the database

Reply:
Indeed. But since this would only be called once ever on a topic, should I save the custom fields in `slack_thread_id=` instead? Since I will never want to call `slack_thread_id=` without following it up, it feels like doing it in `slack_thread_id=` would be an affordance that would avoid future bugs. Any objections there?